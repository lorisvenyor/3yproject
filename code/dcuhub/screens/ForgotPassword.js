
import React from 'react'
import { StyleSheet, Text, View, Image, Dimensions, TextInput, Alert} from 'react-native'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { StatusBar } from 'expo-status-bar'
import * as firebase from 'firebase'
import {SafeAreaView} from 'react-native-safe-area-context'

import Button2 from '../shared/button_2'
import Button1 from '../shared/button_1'

const DeviceHeight = Dimensions.get('screen').height

export default class ForgotPassword extends React.Component{

    state = {
        email: ""
    };
    
    forgotPassword = () => {
        firebase.auth().sendPasswordResetEmail(this.state.email)
          .then(function () {
            Alert.alert('Email Confirmation','Check your email to change your password.')
          }).catch(error => this.setState({errorMessage: error.message}))
        }

    render() {
        return (
            <SafeAreaView style={styles.container}>
            <StatusBar style="auto"/>
            <KeyboardAwareScrollView>
    
                <View style={styles.image_view}>  
                    <Image 
                    source={require("../assets/images/dcuhublogo_higherres.png")}
                    resizeMode="contain"
                    resizeMode="center"
                    style={styles.image_style}>
                    </Image>
                </View> 

                {/* firebase error messages should pop up here */}
                <View style={styles.errorMessage}>
                    {this.state.errorMessage && <Text style={styles.error_txt}>{this.state.errorMessage}</Text>}
                </View>

                <View style={styles.placeholder_style}>
                    <TextInput 
                        style={styles.placeholder_text}
                        placeholder="enter your dcu email address" onChangeText={email => this.setState({email})}
                        value={this.state.email}/> 
                </View>

                <View>
                    <Button2 text='RESET' onPress={this.forgotPassword}/>
                </View>

                <View style={{marginBottom: 30}}>
                    <Button1 text='LOG IN'  onPress={ () => this.props.navigation.navigate('Login')}/>
                </View>

            
            </KeyboardAwareScrollView>
            </SafeAreaView>
        );
    }
}

const styles = StyleSheet.create({
    container : {
        flex: 1,
        height: DeviceHeight,
        backgroundColor: "white"
    },
    placeholder_style: {
        marginHorizontal:40,
        borderWidth: 1.5,
        marginTop: 20,
        paddingHorizontal: 10,
        borderColor:"#25afcd",
        borderRadius: 10,
        paddingVertical: 10 // size of the placeholder
    }, 
    placeholder_text: {
        paddingHorizontal:10
    }, 
    errorMessage: {
        alignItems: "center",
        justifyContent: "center",
        marginHorizontal: 30,
        marginTop: -20,
        marginBottom: 0
    },
    error_txt:{
        color: "#d40059",
        fontSize: 15,
        fontWeight: "600",
        textAlign: "center"
    },
    image_style: {
        width: Dimensions.get('screen').width, 
        height: Dimensions.get('screen').height / 2
    },
    image_view: {
        alignSelf: 'center',
        justifyContent: 'center'
    }
})
