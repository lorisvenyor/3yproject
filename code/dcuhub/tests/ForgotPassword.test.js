import React from 'react'
import {create} from 'react-test-renderer'
import ForgotPassword from '../screens/ForgotPassword'


const screen = create(<ForgotPassword/>)

test('Forgot Password Screen Snapshot', () => {
    expect(screen).toMatchSnapshot()
})
