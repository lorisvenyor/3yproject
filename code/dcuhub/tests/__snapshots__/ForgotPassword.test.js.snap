// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password Screen Snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          resizeMode="center"
          source={1}
          style={
            Object {
              "height": 667,
              "width": 750,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 0,
            "marginHorizontal": 30,
            "marginTop": -20,
          }
        }
      />
      <View
        style={
          Object {
            "borderColor": "#25afcd",
            "borderRadius": 10,
            "borderWidth": 1.5,
            "marginHorizontal": 40,
            "marginTop": 20,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="enter your dcu email address"
          rejectResponderTermination={true}
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "orange",
                "borderRadius": 20,
                "justifyContent": "center",
                "marginHorizontal": 100,
                "marginTop": 30,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              RESET
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 30,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#f4a446",
                "borderRadius": 20,
                "borderWidth": 2,
                "justifyContent": "center",
                "marginHorizontal": 100,
                "marginTop": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#f4a446",
                  "fontSize": 20,
                }
              }
            >
              LOG IN
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
